mathesar_ui/src/components/select/Select.svelte

```diff
@@ -45,7 +45,9 @@
   }
 
   function setSelectedItem() {
-    currentIndex = options.indexOf(value);
+    if (options.some((e) => e[idKey] === value?.[idKey])) {
+      currentIndex = options.indexOf(value);
+    }
```

Comment:
@mr-gabe49 Sorry for not catching this earlier.

The value object need not necessarily be equal to the one in options. They may hold the same values of idKey and valueKey, but be two different objects. So, checking indexOf will return -1, even though we would expect it to return a value.

A better way to do this will be to:

```suggestion
   const index = options.findIndex((e) => e[idKey] === value?.[idKey]);
    if (index > -1) {
      currentIndex = index;
    }
```

Reply:
@pavish Interesting, I'll go ahead and fix it. That's always the case even if the value is set using the options object?

Reply:
@mr-gabe49 No. When the value is set using the options object, it's not the case, because the memory reference for the entries would be the same.

But in some cases, like when the page initially loads (when a filter or group is already applied), we do not set the value from the options. We directly set it based on the url. In such cases, the memory references vary, i.e. the entry within options array would be a different object from the value.

Reply:
Got it, Thanks!